### PR TITLE
Support cache system for Binary Frameworks

### PR DESCRIPTION
**Pull request by giginet**
Currently, Binary Frameworks don't use the cache system.

This PR makes binary frameworks respect cache mode.

### CODE REVIEW

File: Sources/ScipioKit/Producer/BinaryExtractor.swift [hunk 1]
```diff
@@ -8,14 +8,16 @@ struct BinaryExtractor {
...
 
     @discardableResult
-    func extract(of binaryTarget: BinaryTarget) throws -> URL {
-
+    func extract(of binaryTarget: BinaryTarget, overwrite: Bool) throws -> URL {
         let sourcePath = binaryTarget.artifactPath
+        let frameworkName = "\(binaryTarget.c99name).xcframework"
         let fileName = sourcePath.basename
         let destinationPath = outputDirectory.appendingPathComponent(fileName)
-        if !fileSystem.exists(destinationPath) {
```
Comment (giginet) [inline Sources/ScipioKit/Producer/BinaryExtractor.swift:16]: Currently, binary frameworks are never overwritten. So they can't update at all...

**Comment by giginet** [comment]
Added tests 9ce2d52bb15e8870d59c522078fc7c961f85b23d

**Review by giuk-jung — APPROVED** [review]
LGTM

File: Sources/ScipioKit/Producer/BinaryExtractor.swift [hunk 2]
```diff
@@ -8,14 +8,16 @@ struct BinaryExtractor {
...
-            try fileSystem.copy(from: sourcePath.asURL, to: destinationPath)
+        if fileSystem.exists(destinationPath) && overwrite {
```
[the remark this answers is not in the document]
Comment (giginet) [reply Sources/ScipioKit/Producer/BinaryExtractor.swift:16]: I prefer to use `&&` in this context. What do you think?

File: Sources/ScipioKit/Producer/BinaryExtractor.swift [hunk 3]
```diff
@@ -8,14 +8,16 @@ struct BinaryExtractor {
...
+            logger.info("💥 Delete \(frameworkName)", metadata: .color(.red))
```
[the remark this answers is not in the document]
Comment (giginet) [reply Sources/ScipioKit/Producer/BinaryExtractor.swift]: Replaced https://github.com/giginet/Scipio/pull/40/commits/821473c5042eb2580b9b0f3668bb166a84433357

File: Sources/ScipioKit/Producer/BinaryExtractor.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (freddi-kit) [reply Sources/ScipioKit/Producer/BinaryExtractor.swift:16]: Either is fine

File: Sources/ScipioKit/Producer/BinaryExtractor.swift [hunk 4]
```diff
@@ -8,14 +8,16 @@ struct BinaryExtractor {
...
-
+    func extract(of binaryTarget: BinaryTarget, overwrite: Bool) throws -> URL {
         let sourcePath = binaryTarget.artifactPath
+        let frameworkName = "\(binaryTarget.c99name).xcframework"
         let fileName = sourcePath.basename
         let destinationPath = outputDirectory.appendingPathComponent(fileName)
-        if !fileSystem.exists(destinationPath) {
-            try fileSystem.copy(from: sourcePath.asURL, to: destinationPath)
+        if fileSystem.exists(destinationPath) && overwrite {
+            logger.info("🗑️ Delete \(frameworkName)", metadata: .color(.red))
```
Comment (freddi-kit) [inline Sources/ScipioKit/Producer/BinaryExtractor.swift:17]: orange or yellow is better? same reason with emoji of error

**Review by freddi-kit — APPROVED** [review]

**Pull request merged**